core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -82,9 +84,13 @@
   @GuardedBy("this")
   private Listener listener;
 
-  DnsNameResolver(@Nullable String nsAuthority, String name, Attributes params,
+  DnsNameResolver(
+      @Nullable String nsAuthority,
+      String name,
+      Attributes params,
       Resource<ScheduledExecutorService> timerServiceResource,
-      Resource<ExecutorService> executorResource) {
+      Resource<ExecutorService> executorResource,
+      boolean balancerLookup) {
```

Comment:
Although SRV is not grpclb specific, that SRV is used for balancer lookup (as this boolean argument implies) is grpclb specific. Consider changing it to "srvLookup" and a generic SRV plumbing path (e.g., as attributes) that don't have any "balancer" indication.